From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sun, 10 Jan 2016 20:40:33 -0600
Subject: [PATCH] Disable chest cat detection


diff --git a/src/main/java/net/minecraft/server/BlockChest.java b/src/main/java/net/minecraft/server/BlockChest.java
index f725aea8cf2e80a0014868c3fd31b0f794f65efc..eed516e223607e84bb475c51cef21e7f0b1073c2 100644
--- a/src/main/java/net/minecraft/server/BlockChest.java
+++ b/src/main/java/net/minecraft/server/BlockChest.java
@@ -410,6 +410,11 @@ public class BlockChest extends BlockContainer {
     }
 
     private boolean p(World world, BlockPosition blockposition) {
+        // PaperSpigot start - Option to disable chest's cat detection (Performance++)
+        if (world.paperSpigotConfig.disableChestCatDetection) {
+            return false;
+        }
+        // PaperSpigot end
         Iterator iterator = world.a(EntityOcelot.class, new AxisAlignedBB((double) blockposition.getX(), (double) (blockposition.getY() + 1), (double) blockposition.getZ(), (double) (blockposition.getX() + 1), (double) (blockposition.getY() + 2), (double) (blockposition.getZ() + 1))).iterator();
 
         EntityOcelot entityocelot;
diff --git a/src/main/java/org/github/paperspigot/PaperSpigotWorldConfig.java b/src/main/java/org/github/paperspigot/PaperSpigotWorldConfig.java
index 7eb34630ef6b7ffddad2e02513d2f4652bdfaa66..b6ce3bbea2a73b94f16a1cab8c2846f4a5b302ef 100644
--- a/src/main/java/org/github/paperspigot/PaperSpigotWorldConfig.java
+++ b/src/main/java/org/github/paperspigot/PaperSpigotWorldConfig.java
@@ -193,10 +193,12 @@ public class PaperSpigotWorldConfig
 
     public boolean boatsDropBoats;
     public boolean disablePlayerCrits;
+    public boolean disableChestCatDetection;
     private void mechanicsChanges()
     {
         boatsDropBoats = getBoolean( "game-mechanics.boats-drop-boats", false );
         disablePlayerCrits = getBoolean( "game-mechanics.disable-player-crits", false );
+        disableChestCatDetection = getBoolean( "game-mechanics.disable-chest-cat-detection", false );
     }
 
     public boolean netherVoidTopDamage;
