From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: kashike <kashike@vq.lc>
Date: Tue, 16 Feb 2016 13:36:04 -0800
Subject: [PATCH] Be a bit more informative in maxHealth exception


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
index 5317cff66c582307c307589f545c64fea87769cc..3e3a04808196e722fd1dcdae638bfcf45a79f4d9 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
@@ -80,7 +80,9 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
 
     public void setHealth(double health) {
         if ((health < 0) || (health > getMaxHealth())) {
-            throw new IllegalArgumentException("Health must be between 0 and " + getMaxHealth());
+            throw new IllegalArgumentException("Health must be between 0 and " + getMaxHealth() + ", but was " + health
+                + ". (attribute base value: " + this.getHandle().getAttributeInstance(GenericAttributes.maxHealth).b()
+            + (this instanceof CraftPlayer ? ", player: " + this.getName() + ')' : ')'));
         }
 
         if (entity instanceof EntityPlayer && health == 0) {
