From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 6 Nov 2014 18:29:20 -0600
Subject: [PATCH] Add async chunk load API


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 2772b09e3eb2121f267ccd9fd2e3288ade1c6f1a..325d65a37dc5e38bf64cb88b936d2a258be182ab 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -129,6 +129,15 @@ public interface World extends PluginMessageRecipient, Metadatable {
      */
     public Chunk getChunkAt(Block block);
 
+    // PaperSpigot start - Async chunk load API
+    public static interface ChunkLoadCallback {
+        public void onLoad(Chunk chunk);
+    }
+    public void getChunkAtAsync(int x, int z, ChunkLoadCallback cb);
+    public void getChunkAtAsync(Location location, ChunkLoadCallback cb);
+    public void getChunkAtAsync(Block block, ChunkLoadCallback cb);
+    // PaperSpigot end
+
     /**
      * Checks if the specified {@link Chunk} is loaded
      *
