From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniele Bruni <67119358+danib150@users.noreply.github.com>
Date: Mon, 19 Jun 2023 18:01:15 +0200
Subject: [PATCH] Started editing About command


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 8fbc82833d16138914e379fef9a38c206d9cc07b..868740b8e9f015a6205f509646955d0d6bd3f8cc 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -76,7 +76,7 @@ public final class Bukkit {
         }
 
         Bukkit.server = server;
-        server.getLogger().info("This server is running " + getName() + " version " + getVersion() + " (Implementing API version " + getBukkitVersion() + ")");
+        server.getLogger().info(ChatColor.RED + "This server is running " + getName() + " version " + getVersion() + " (Implementing API version " + getBukkitVersion() + ")");
     }
 
     /**
diff --git a/src/main/java/org/bukkit/command/defaults/VersionCommand.java b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
index e91e1d8c1215d19a78611d3c8302ed9634bfbda5..ff721c424289eac2cf78e7094d65509367dcd79b 100644
--- a/src/main/java/org/bukkit/command/defaults/VersionCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
@@ -43,7 +43,9 @@ public class VersionCommand extends BukkitCommand {
         if (!testPermission(sender)) return true;
 
         if (args.length == 0) {
-            sender.sendMessage("This server is running " + Bukkit.getName() + " version " + Bukkit.getVersion() + " (Implementing API version " + Bukkit.getBukkitVersion() + ")");
+            sender.sendMessage(
+                    ChatColor.DARK_GRAY + "> " + ChatColor.WHITE + "This server is running " + ChatColor.GREEN + Bukkit.getName()
+                            + "\n" + ChatColor.DARK_GRAY + ">" + ChatColor.WHITE + " version " + ChatColor.GREEN + Bukkit.getVersion() + " §8(§fImplementing API version " + ChatColor.GREEN + Bukkit.getBukkitVersion() + "§8)");
             sendVersion(sender);
         } else {
             StringBuilder name = new StringBuilder();
